Merging from perfview->perfviewJS to PerfviewJS

murugaratham opened the pull request:
@colonelchlorine 

With my changes for react & electron

brianrob commented:
@murugaratham thanks much for the contribution.  This looks pretty big - I'd like to understand what all is included here, and whether or not we should be breaking this up into multiple PRs.  Can you share a list of the changes here?

brianrob commented:
Thanks @murugaratham.  I will review and report back.

brianrob commented:
@murugaratham, I just want to acknowledge that you've done a lot of work here to clean-up the UI and make it much more pleasant.  Thank you!

I have a few follow-up questions:
1. Can you share details about where the images came from?  I want to make sure that we're OK to include these.
2. I'm noticing that the default port number that the front-end uses to communicate with the backend has changed.  Is this intentional?
3. Does the theme switcher at the top right work for you?  I tried it locally here, but it doesn't seem to work.
4. I tried dragging and dropping a file into the viewer, but this failed with this:
<img width="794" alt="image" src="https://user-images.githubusercontent.com/6210322/153945935-5295b65e-f987-467a-93f8-559c5eb390bc.png">

I may have more as I keep experimenting, but this is it for now.

murugaratham commented:
Hi @brianrob 

1. Icons are from  
   1. @fortawesome/free-solid-svg-icons
   1. @fortawesome/free-brands-svg-icons
   1. @fluentui/react/lib/Icons

Font-awesome free icons, so i guess it should be fine, here's their license https://github.com/FortAwesome/Font-Awesome/blob/6.x/LICENSE.txt

1. I'm doing this on Mac, and Apple had hijacked 5000 from us, [stackoverflow](https://stackoverflow.com/a/69829313/670514) answered this too, so i changed to 6000 instead as default port, i reckon, this isn't biggie

1. I made a change to use react-hookz for useLocalStorageValue, which will "emit" change, previous hook only set value within localStorage, but "subscribers" aren't getting updated values

1. I've updated the `launch.json` to use `npm run dev` instead of `npm run start`, previously @mjsabby suggested to use electron. With electron, it should work fine
